### ISSUE TITLE
`StringUtils.collectionToDelimitedString(…)` fails with `NullPointerException` when the collection contains `null`

### ISSUE THREAD

**Issue by mp911de** [spring-projects/spring-framework#27411]
**Affects:** 5.3.10

After #26316, `StringUtils.collectionToDelimitedString(…)` behaves slightly different than the previous implementation in 5.3.9.

Calling `StringUtils.collectionToDelimitedString(Collections.singleton(null), ",")` in version 5.3.10 fails with:

```
java.lang.NullPointerException
	at org.springframework.util.StringUtils.collectionToDelimitedString(StringUtils.java:1304)
	at org.springframework.util.StringUtils.collectionToDelimitedString(StringUtils.java:1326)
```

whereas version 5.3.9 returned the string `"null"`.

This ticket is to clarify which behavior is the correct one.

See also spring-projects/spring-data-redis#2167

**Comment by bclozel** [comment]
Closing in favour of #27419
